src/tests/gov/nasa/jpf/test/java/lang/StackWalkerTest.java

```diff
@@ -68,4 +67,79 @@ public void testCallerClass() {
       callIt();
     }
   }
+
+  void reflectionCallee() {
+    // Case 1, don't show reflect frames
+    // By default, reflect frames are filtered out
+    final StackWalker walker1 = StackWalker.getInstance();
+    List<String> methodNames1 = walker1.walk(s -> s
+        .map(StackWalker.StackFrame::getMethodName)
+        .collect(Collectors.toList()));
+    assertTrue(methodNames1.get(0).equals("reflectionCallee"));
+    assertTrue(methodNames1.get(1).equals("testShowReflectFrame"));
+
+    // Case 2, show reflect frames
+    final StackWalker walker2 = StackWalker.getInstance(StackWalker.Option.SHOW_REFLECT_FRAMES);
+    List<String> methodNames2 = walker2.walk(s -> s
+        .map(StackWalker.StackFrame::getMethodName)
+        .collect(Collectors.toList()));
+    // Since implementation of reflect invocation is JVM's internal detail and
+    // could be different across different JVMs. We need to loose the constraints.
+    // But there should be at least two `Method::invoke`s on stack (One for
+    // reflectionCallee() reflection call and another for unit test reflection call).
+    System.out.println(methodNames2);
+    assertTrue(Collections.frequency(methodNames2, "invoke") >= 2);
+  }
+
+  @Test
+  public void testShowReflectFrame() throws Exception {
+    if (verifyNoPropertyViolation()){
+      Method reflectCallee = StackWalkerTest.class.getDeclaredMethod("reflectionCallee");
+      reflectCallee.invoke(new StackWalkerTest());
+    }
+  }
+
+  @Test
+  public void testStackWalkerInCaseOfChoiceGenerator() {
+    if (verifyNoPropertyViolation()){
+      Thread t1 = new Thread() {
+        @Override
+        public void run() {
+          callIt();
+        }
+      };
+      Thread t2 = new Thread() {
+        @Override
+        public void run() {
+          callIt();
+        }
+      };
+      t1.start();
+      t2.start();
+    }
+  }
+
+  void deepCallRecur(int n, int depth) {
+    if (n == 1) {
+      final StackWalker walker = StackWalker.getInstance(StackWalker.Option.RETAIN_CLASS_REFERENCE);
+      List<String> methods = walker.walk(s -> s
+          .map(StackWalker.StackFrame::getMethodName)
+          .filter(m -> m.equals("deepCallRecur"))
+          .collect(Collectors.toList()));
+      assertEquals(methods.size(), depth);
+    } else {
+      deepCallRecur(n - 1, depth);
+    }
+  }
+
+  void deepCall(int depth) {
+    deepCallRecur(depth, depth);
+  }
+
+  @Test
+  public void testDeepStack() {
+    if (verifyNoPropertyViolation()) {
+      deepCall(100);
```

Comment:
Decrease this to 5; it is unlikely that this functionality behaves differently for larger depths, and we want to keep the tests small and efficient.